solo-io/gloo
# gateway2/deployer: fix race due to context cancellation

# Description

Avoid a helm race condition in our deployer code.



# Context
In https://github.com/solo-io/gloo/pull/9646 we run our tests with `-race`. In doing so, we encountered a race condition: https://github.com/solo-io/solo-projects/issues/6408.

Upon speaking with @yuval-k ([slack ref](https://solo-io-corp.slack.com/archives/C06C8RA01NF/p1718893489626729)), the proposed plan is to side step the race condition in helm.

## Interesting decisions
-

## Testing steps
- This should not change any existing behavior, so I intentionally did not change tests. 

## Notes for reviewers
-

# Checklist:
